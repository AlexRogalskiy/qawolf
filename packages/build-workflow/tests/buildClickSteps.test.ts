import { loadEvents } from "@qawolf/fixtures";
import { buildClickSteps } from "../";

describe("buildClickSteps", () => {
  let loginEvents: any[];
  let clickInputEvents: any[];

  beforeAll(async () => {
    loginEvents = await loadEvents("scroll_login");
    clickInputEvents = await loadEvents("click_input");
  });

  it("builds expected steps", async () => {
    const steps = buildClickSteps(loginEvents);
    expect(steps).toMatchSnapshot();
  });

  it("builds a click step on an input if it is not followed by a type event", async () => {
    const steps = buildClickSteps(clickInputEvents);
    expect(steps).toMatchSnapshot();
  });
});
