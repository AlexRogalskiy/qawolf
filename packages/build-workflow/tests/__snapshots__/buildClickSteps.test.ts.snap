// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`buildClickSteps builds a click step on an input if it is not followed by a type event 1`] = `
Array [
  Object {
    "action": "click",
    "html": Object {
      "ancestors": Array [
        Object {
          "attrs": Object {
            "class": "large-6 small-12 columns",
          },
          "children": Array [],
          "name": "div",
          "type": "tag",
          "voidElement": false,
        },
        Object {
          "attrs": Object {
            "class": "row",
          },
          "children": Array [],
          "name": "div",
          "type": "tag",
          "voidElement": false,
        },
      ],
      "node": Object {
        "attrs": Object {
          "id": "username",
          "name": "username",
          "type": "text",
        },
        "children": Array [],
        "name": "input",
        "type": "tag",
        "voidElement": true,
      },
    },
    "index": 0,
    "page": 0,
  },
]
`;

exports[`buildClickSteps builds expected steps 1`] = `
Array [
  Object {
    "action": "click",
    "html": Object {
      "ancestors": Array [
        Object {
          "attrs": Object {},
          "children": Array [],
          "name": "li",
          "type": "tag",
          "voidElement": false,
        },
        Object {
          "attrs": Object {},
          "children": Array [],
          "name": "ul",
          "type": "tag",
          "voidElement": false,
        },
      ],
      "node": Object {
        "attrs": Object {
          "href": "/login",
        },
        "children": Array [
          Object {
            "content": "Form Authentication",
            "type": "text",
          },
        ],
        "name": "a",
        "type": "tag",
        "voidElement": false,
      },
    },
    "index": 11,
    "page": 0,
  },
  Object {
    "action": "click",
    "html": Object {
      "ancestors": Array [
        Object {
          "attrs": Object {
            "class": "example",
          },
          "children": Array [],
          "name": "div",
          "type": "tag",
          "voidElement": false,
        },
        Object {
          "attrs": Object {
            "class": "large-12 columns",
            "id": "content",
          },
          "children": Array [],
          "name": "div",
          "type": "tag",
          "voidElement": false,
        },
      ],
      "node": Object {
        "attrs": Object {
          "class": "button secondary radius",
          "href": "/logout",
        },
        "children": Array [
          Object {
            "attrs": Object {
              "class": "icon-2x icon-signout",
            },
            "children": Array [
              Object {
                "content": " Logout",
                "type": "text",
              },
            ],
            "name": "i",
            "type": "tag",
            "voidElement": false,
          },
        ],
        "name": "a",
        "type": "tag",
        "voidElement": false,
      },
    },
    "index": 39,
    "page": 0,
  },
]
`;
